static/app/views/performance/transactionSummary/transactionOverview/content.tsx

```diff
@@ -363,6 +359,7 @@ function SummaryContent({
           searchSource="transaction_summary"
           disableLoadingTags // already loaded by the parent component
           filterKeyMenuWidth={420}
+          trailingItems={generateTrailingItems()}
```

Comment:
This use of `useCallback` doesn't really make sense as you're not passing the memorized callback anywhere. It could make more sense if you used `useMemo` to memorize the component but even then, you could do something as simple as to achieve the same effect.
```suggestion
          trailingItems={canUseTransactionMetricsData(organization, mepDataContext) ? null : <MetricsWarningIcon />}
```
Not sure about the implementation of `canUseTransactionMetricsData` but if that's expensive, you can cache that and do something like
```tsx
  const ableToUseTransactionMetricsData = canUseTransactionMetricsData(organization, mepDataContext);
  return (
        <TransactionSearchQueryBuilder
          ...
          trailingItems={ableToUseTransactionMetricsData ? null : <MetricsWarningIcon />}
```

Reply:
ah yes good catch, I'll memoize the trailing items instead. It's not that `canUseTransactionMetricsData`, but I'm trying to avoid unnecessary rerenders on the `SearchQueryBuilder` component, as I've been warned that rerenders are expensive for it.